### DISCUSSION TITLE
classes/class_node3d

### DISCUSSION THREAD

**Discussion by giscus** [godotengine/godot-docs-user-notes#283] [User-contributed notes]
# classes/class_node3d

Inherits: Node< Object Inherited By: AudioListener3D, AudioStreamPlayer3D, BoneAttachment3D, Camera3D, CollisionObject3D, CollisionPolygon3D, CollisionShape3D, GridMap, ImporterMeshInstance3D, Join...

https://docs.godotengine.org/en/stable/classes/class_node3d.html

**Comment by Bogost** [comment]
Methods like look_at() wont work unless node is in scene tree (if you are adding nodes from script you thirst need to use add_child() and then look_at() )

**Comment by dagarsar** [comment]
In ```CanvasItem```'s docs, the ```is_visible_in_tree``` method contains this note about the visibility inheritance:

> Visibility is checked only in parent nodes that inherit from CanvasItem, CanvasLayer, and Window. If the parent is of any other type (such as Node, AnimationPlayer, or Node3D), it is assumed to be visible.

I think the equivalent method in this class should have the same note.

**Reply by Calinou** [reply]
Pull request opened:

- https://github.com/godotengine/godot/pull/102450

**Comment by NatCracken** [comment]
For anyone else wondering about Node3D.quaternion; its local. What you're probably looking for if ur using quats is the equivalent of what would be "Node3D.global_quaternion", which you can get with Node3D.global_basis.get_rotation_quaternion(). If u want to set it you will need to use Node3D.global_basis with the quaternion constructor. Don't forget to normalize regularly and good luck out there!

**Comment by NoahJacobsen** [comment]
Per the [Physics Introduction tutorial](https://docs.godotengine.org/en/stable/tutorials/physics/physics_introduction.html#using-rigidbody2d), unexpected behaviour can occur when trying to alter the `position` or other physics properties, when parenting a `RigidBody3D`. In my case, while the physics object was in motion, the new position was overwritten by the previous position once the `_physics_process()` was reached.

A workaround could be to queue the desired position (or other physics property value), and then empty the queue at the beginning of the RigidBody's `_physics_process()`.
